src/pages/components/CharacterCountDown/CharacterCountDown.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+
+interface ICharacterCountdownProps {
+  maxLength: number;
+}
+
+const CharacterCountdown: React.FC<ICharacterCountdownProps> = ({
+  maxLength,
+}) => {
+  const [text, setText] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const newText = event.target.value;
+    if (newText.length <= maxLength) {
+      setText(newText);
+    }
+  };
+
+  return (
+    <div className="">
+      <textarea
+        value={text}
+        onChange={handleChange}
+        maxLength={maxLength}
+        placeholder={`Digite até ${maxLength} caracteres`}
+      />
+      <p>{maxLength - text.length} caracteres restantes</p>
+    </div>
```

Comment:
Não precisa da div aqui, use um React Fragment ao invés da div.

O fragment não gera item no DOM, fazendo com que fique melhor a perfomance desse componente.
https://react.dev/reference/react/Fragment

```suggestion
    <>
      <textarea
        value={text}
        onChange={handleChange}
        maxLength={maxLength}
        placeholder={`Digite até ${maxLength} caracteres`}
      />
      <p>{maxLength - text.length} caracteres restantes</p>
    <>
```